### ISSUE TITLE
2 Apollo on the same Simualtion

### ISSUE THREAD

**Issue by gmvc2000** [ApolloAuto/apollo#15290]
I'm running 2 autonomous vehicles on the same simulation to test a cooperative autonomous driving scenario. But I can't understand if those vehicles can see each other or not and stop if one vehicle is in front of the other

**Comment by daohu527** [comment]
Self-driving cars obtain surrounding environment information through perception modules. That is to say, the perception module will detect vehicles and pedestrians and provide feedback to the self-driving car to avoid them.

Back to your question, you can parse the autonomous car's perception messages(topic: /apollo/perception/obstacles) to see if another car is detected. 

## go further
I think you may need to identify it automatically. You can use the simulator to obtain the position of the self-driving vehicle, and then compare it with the perceived results. For example, use IOU to determine whether it is the same car.

**Comment by YuqiHuai** [comment]
https://github.com/Software-Aurora-Lab/DoppelTest This project allows multiple Apollo instances to be simulated in SimControl. The underlying mechanism passes one Apollo's localization as perception to other instances, so all the instances can see each other.

**Comment by daohu527** [comment]
> https://github.com/Software-Aurora-Lab/DoppelTest This project allows multiple Apollo instances to be simulated in SimControl. The underlying mechanism passes one Apollo's localization as perception to other instances, so all the instances can see each other.

So are there no other NPCs? All vehicles are Apollo self-driving cars?

**Comment by YuqiHuai** [comment]
> > https://github.com/Software-Aurora-Lab/DoppelTest This project allows multiple Apollo instances to be simulated in SimControl. The underlying mechanism passes one Apollo's localization as perception to other instances, so all the instances can see each other.
> 
> So are there no other NPCs? All vehicles are Apollo self-driving cars?

Not exactly. In this implementation, all cars are Apollos, pedestrian walk around crosswalks (didn’t have a better way to simulate realistic pedestrian, no bicycle, with valid traffic lights (conflicting ones won’t be green at the same time).

**Comment by daohu527** [comment]
What I want to express is that there will be conflicts between groundtruth and perception modules. It is best for them to choose 1 from 2.

**Comment by YuqiHuai** [comment]
I see. Yes, in the project linked above, perception module is replaced by ground truth perception messages generated by the project.
